e2e/nomostest/git-server.go

```diff
@@ -127,9 +127,6 @@ func gitDeployment() *appsv1.Deployment {
 		core.Namespace(testGitNamespace),
 		core.Labels(testGitServerSelector()),
 	)
-	if *e2e.GKEAutopilot {
-		deployment.SetAnnotations(map[string]string{safeToEvictAnnotation: "false"})
```

Comment:
Note: this was setting the annotation on the deployment rather than the pod template